internal/resources/metal/helper/regex.go

```diff
@@ -0,0 +1,24 @@
+package helper
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+)
+
+func StringToRegex(pattern string) (regExp *regexp.Regexp) {
```

Comment:
I don't think we should write wrapper functions for this.  The resources should continue to define validation directly, like `stringvalidator.RegexMatches(regexp.MustCompile(<pattern>), <message>)` rather than using a wrapper.